docs/graphql/manual/hasura-cli/reference.rst

```diff
@@ -0,0 +1,85 @@
+.. meta::
+   :description: Haura GarphQL CLI configuration reference 
+   :keywords: hasura, docs, CLI, flags, config 
+
+Hasura CLI configuration reference 
+==================================
+
+.. contents:: Table of contents
+  :backlinks: none
+  :depth: 1
+  :local:
+
+Hasura CLI commands get key values from flags, ENV variables, ``.env`` file and configuration file ``config.yaml``. 
+
+Configuration file format
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The configuration file, ``config.yaml`` format with default values is given below.
+
+.. code-block:: none
+
+  version: 1
+  endpoint: http://localhost:8080
+  admin_secret:
```

Comment:
Can we also add if a field is optional and the config version from which it is supported?

Reply:
Included a table for the above, similar to [this](https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/index.html#query).